### PR TITLE
Dev branch

### PR DESCRIPTION
**Pull request by athilkabbas123**
**Summary**

- Adding functionalities for the following endpoints
-       1. /order/{id}/submission/{submissionId}
-       2. /order/{id}/submission   
-       3./order/{id}/rejection/{rejectionId}
-       4./order/{id}/rejection
-       5./order/{id}/amendment/{amendmentId}
-       6./order/{id}/amendment
-       7./order/{id}/acceptance/{acceptanceId}
-       8./order/{id}/acceptance
 
- Adding field called 'type' which takes values Submission, Rejection, Acceptance, Amendment for the corresponding actions while inserting into order_transaction table

- Adding validation for all order get endpoints which required id

### CODE REVIEW

File: app/db/index.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ async function postOrderDb(reqBody) {
       supplier: reqBody.supplierAddress,
       required_by: reqBody.requiredBy,
       items: reqBody.items,
-      buyer: reqBody.buyerAddress,
```
Comment (sownak) [inline app/db/index.js:24]: why is this key changed?

File: app/db/index.js [hunk 2]
```diff
@@ -74,6 +74,16 @@ async function getOrder(id) {
   return client('orders').select().where({ id })
 }
 
+async function getOrderTransactions(order_id,type){
+  return client('order_transactions').select().where({order_id, type})
+}
+async function getOrderTransactionsById(id,order_id,type){
```
Comment (sownak) [inline app/db/index.js:80]: confusion between id and order_id, maybe rename them properly.

**Review by sownak — CHANGES_REQUESTED** [review]

**Comment by sownak** [comment]
Raise PR to develop branch

File: app/db/index.js (hunk 1 above)
Comment (athilkabbas123) [reply app/db/index.js:24]: In order table schema the key is purchaser not buyer

**Pull request closed without merging**